src/main/java/com/libertymutualgroup/herman/aws/ecs/HermanContainerDefinition.java

```diff
@@ -0,0 +1,318 @@
+package com.libertymutualgroup.herman.aws.ecs;
+
+import com.amazonaws.internal.SdkInternalList;
+import com.amazonaws.services.ecs.model.ContainerDefinition;
+import com.amazonaws.services.ecs.model.HealthCheck;
+import com.amazonaws.services.ecs.model.HostEntry;
+import com.amazonaws.services.ecs.model.KeyValuePair;
+import com.amazonaws.services.ecs.model.LinuxParameters;
+import com.amazonaws.services.ecs.model.LogConfiguration;
+import com.amazonaws.services.ecs.model.MountPoint;
+import com.amazonaws.services.ecs.model.PortMapping;
+import com.amazonaws.services.ecs.model.RepositoryCredentials;
+import com.amazonaws.services.ecs.model.Ulimit;
+import com.amazonaws.services.ecs.model.VolumeFrom;
+
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+public class HermanContainerDefinition {
+
+    private String name;
+    private String image;
+    private RepositoryCredentials repositoryCredentials;
+    private Integer cpu;
+    private Integer memory;
+    private Integer memoryReservation;
+    private SdkInternalList<String> links;
+    private SdkInternalList<PortMapping> portMappings;
+    private Boolean essential;
+    private SdkInternalList<String> entryPoint;
+    private SdkInternalList<String> command;
+    private List<EnvironmentVariableDefinition> environment;
+    private SdkInternalList<MountPoint> mountPoints;
+    private SdkInternalList<VolumeFrom> volumesFrom;
+    private LinuxParameters linuxParameters;
+    private String hostname;
+    private String user;
+    private String workingDirectory;
+    private Boolean disableNetworking;
+    private Boolean privileged;
+    private Boolean readonlyRootFilesystem;
+    private SdkInternalList<String> dnsServers;
+    private SdkInternalList<String> dnsSearchDomains;
+    private SdkInternalList<HostEntry> extraHosts;
+    private SdkInternalList<String> dockerSecurityOptions;
+    private Map<String, String> dockerLabels;
+    private SdkInternalList<Ulimit> ulimits;
+    private LogConfiguration logConfiguration;
+    private HealthCheck healthCheck;
+
+    public ContainerDefinition toEcsContainerDefinition() {
+        return new ContainerDefinition()
```

Comment:
Had to do it this way because AWS owns the ContainerDefinition and KeyValuePair objects, making it hard to extend in order to add the "encrypt" value to the environment variable definition. This way we can extend ECS's definition.

Reply:
I'm not a big fan of this because it creates more overhead when ContainerDefinition is modified